include/mbedtls/aes.h

```diff
@@ -70,6 +70,17 @@
 extern "C" {
 #endif
 
+#if !defined(MBEDTLS_DEPRECATED_REMOVED)
+#if defined(MBEDTLS_DEPRECATED_WARNING)
+#define MBEDTLS_DEPRECATED      __attribute__((deprecated))
+#else
+#define MBEDTLS_DEPRECATED
+#endif
+
+#define MBEDTLS_ERR_AES_FEATURE_UNAVAILABLE static inline int MBEDTLS_DEPRECATED( void ){ return( MBEDTLS_ERR_PLATFORM_FEATURE_UNSUPPORTED ); }
```

Comment:
- What's the name of this inline function?
- Why do we need an inline function? Add a comment.
- I think I'd rather update our test scripts to work with constant deprecation than to use inline functions to work around the scripts' limitations.

Reply:
>What's the name of this inline function?
Copy paste error. I'll add names to the functions

>Why do we need an inline function? Add a comment.

I couldn't find another way to deprecate MACROs. If I add a warning, we get compilation warnings, and thus compilation errors when warnings are treated as errors. see https://stackoverflow.com/questions/2681259/how-to-deprecate-a-macro-in-gcc

>I think I'd rather update our test scripts to work with constant deprecation than to use inline functions to work around the scripts' limitations.

Me too, but how do you suggest we make the constant deprecations?

Reply:
Note that `all.sh` checks explicitely for `MBEDTLS_DEPRECATED_WARNINGS` and treat warnings as errors:
~~~
msg "build: make, full config + DEPRECATED_WARNING, gcc -O" # ~ 30s
cleanup
cp "$CONFIG_H" "$CONFIG_BAK"
scripts/config.pl full
scripts/config.pl set MBEDTLS_DEPRECATED_WARNING
# Build with -O -Wextra to catch a maximum of issues.
make CC=gcc CFLAGS='-O -Werror -Wall -Wextra' lib programs
make CC=gcc CFLAGS='-O -Werror -Wall -Wextra -Wno-unused-function' tests

msg "build: make, full config + DEPRECATED_REMOVED, clang -O" # ~ 30s
# No cleanup, just tweak the configuration and rebuild
make clean
scripts/config.pl unset MBEDTLS_DEPRECATED_WARNING
scripts/config.pl set MBEDTLS_DEPRECATED_REMOVED
# Build with -O -Wextra to catch a maximum of issues.
make CC=clang CFLAGS='-O -Werror -Wall -Wextra' lib programs
make CC=clang CFLAGS='-O -Werror -Wall -Wextra -Wno-unused-function' tests
~~~

Reply:
@Patater How about we go with our original plan and define the MACROS as the platform errors, but not using the `MBEDTLS_DEPRECATED_WARNING`?